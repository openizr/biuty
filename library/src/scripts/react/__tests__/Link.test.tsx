/**
 * Copyright (c) Matthieu Jabbour. All Rights Reserved.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 *
 */

import React from 'react';
import UILink from 'scripts/react/Link';
import { act } from 'react-dom/test-utils';
import { render, unmountComponentAtNode } from 'react-dom';

let container = document.createElement('div');

describe('react/UILink', () => {
  beforeEach(() => {
    container = document.createElement('div');
    document.body.appendChild(container);
    jest.clearAllMocks();
  });

  afterEach(() => {
    unmountComponentAtNode(container);
    container.remove();
    ((container as unknown) as null) = null;
  });

  test('renders correctly - basic', () => {
    act(() => {
      render(<UILink label="Test" href="https://test.com" modifiers="large" />, container);
    });
    expect(container).toMatchSnapshot();
  });

  test('renders correctly - with id', () => {
    act(() => {
      render(<UILink id="test" label="Test" href="https://test.com" />, container);
    });
    const link = document.querySelector('a') as HTMLAnchorElement;
    act(() => {
      link.dispatchEvent(new MouseEvent('click', { bubbles: true }));
    });
    expect(container).toMatchSnapshot();
  });

  test('renders correctly - with listener', () => {
    const onClick = jest.fn();
    act(() => {
      render(<UILink onClick={onClick} label="Test" href="https://test.com" />, container);
    });
    const link = document.querySelector('a') as HTMLAnchorElement;
    act(() => {
      link.dispatchEvent(new MouseEvent('click', { bubbles: true }));
    });
    expect(container).toMatchSnapshot();
    expect(onClick).toHaveBeenCalledTimes(1);
  });
});
