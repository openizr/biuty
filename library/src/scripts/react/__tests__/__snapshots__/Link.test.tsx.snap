// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`react/UILink renders correctly - basic 1`] = `
<a
  class="ui-link ui-link--large"
  href="https://test.com"
>
  Test
</a>
`;

exports[`react/UILink renders correctly - with id 1`] = `
<a
  class="ui-link"
  href="https://test.com"
  id="test"
>
  Test
</a>
`;

exports[`react/UILink renders correctly - with itemProp 1`] = `
<a
  class="ui-link"
  href="https://test.com"
>
  Test
</a>
`;

exports[`react/UILink renders correctly - with rel 1`] = `
<a
  class="ui-link"
  href="https://test.com"
  rel="no referrer"
>
  Test
</a>
`;

exports[`react/UILink renders correctly - with target 1`] = `
<a
  class="ui-link"
  href="https://test.com"
  target="_blank"
>
  Test
</a>
`;

exports[`react/UILink renders correctly - with title 1`] = `
<a
  class="ui-link"
  href="https://test.com"
  title="test"
>
  Test
</a>
`;
